### ISSUE TITLE
Can't save number with leading zero to DC_File

### ISSUE THREAD

**Issue by christian-kolb** [contao/core#4035]
I have a DCA File which uses "File" as DataContainer.

```
// Config
'config' => array
(
    'dataContainer'               => 'File',
    'closed'                      => true
)
```

One of the fields is there to save the postal which can be a number with a leading zero.

``` php
'postal' => array
(
    'label'                   => &$GLOBALS['TL_LANG']['settings']['postal'],
    'inputType'               => 'text',
    'exclude'                 => true,
    'eval'                    => array('mandatory'=>true, 'maxlength'=>32, 'tl_class'=>'w50'),
),
```

Unfortunately it doesn't work. Whenever I insert a number with a leadings zero, the value gets changed.

For example: "01129" is transformed to "74".
When I insert a letter at the end like "01129A" everything works like a charm. But there shouldn't be a letter at the end of a postal.

This works fine when I save the values to a table.

**Comment by leofeyer** [comment]
We have discussed this in Mumble on January 26th and we have agreed that this is a known limitation. The `localconfig.php` is to be deprecated in Contao 4 anyways, therefore we do not want to put time and energy into this issue.

**Comment by ausi** [comment]
Sorry to bring this up again. The localconfig is still used in Contao 4 and the issue can be reproduced with the core fields (e.g. entering `0123` in *“Exclude folders from synchronization”*).

I think this can be easily fixed by changing the regex in [Config.php:475](https://github.com/contao/core/blob/5e2f2662b79755ccde5ffb6cb3660462cb7e64ec/system/modules/core/library/Contao/Config.php#L475) to `!preg_match('/e|^[+-]?0[^.]/', $varValue)`.

**Comment by discordier** [comment]
Won't this then change existing behaviour in an incompatible way?
I mean, existing values will then get resaved differently (see comments above)?

IMO the only safe way would be to "type hint" within the field eval, as pointed out above by @dmolineus, yet this would be a new feature.

**Comment by ausi** [comment]
> Won't this then change existing behaviour in an incompatible way?

I don’t think so.

> I mean, existing values will then get resaved differently (see comments above)?

Why? Can you give an example?

**Comment by discordier** [comment]
You are right, it will still safe every fractional number as string. My concerns proved invalid.

**Comment by leofeyer** [comment]
Fixed in f690d69997b7d7bfb5b936d9e99e3a2fd2311d6a then. Thank you @ausi.
